### PR TITLE
♻️ Refactor/remove liquidator from factory

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/Factory.sol [hunk 1]
```diff
@@ -35,7 +35,7 @@ contract Factory is ERC721, Ownable {
 
     string public baseURI;
 
-    address public liquidatorAddress;
+    //address public liquidatorAddress;
```
Comment (Thomas-Smets) [inline src/Factory.sol:38]: Remove variable

File: src/Factory.sol [hunk 2]
```diff
@@ -59,10 +59,10 @@ contract Factory is ERC721, Ownable {
     @dev Since vaults to be liquidated, together with the open debt, are transferred to the protocol,
          New logic can be set without needing to increment the vault version.
     @param _newLiquidator The new liquidator contract
-  */
     function setLiquidator(address _newLiquidator) public onlyOwner {
```
Comment (Thomas-Smets) [inline src/Factory.sol:62]: Remove function

File: src/Factory.sol [hunk 3]
```diff
@@ -359,17 +359,20 @@ contract Factory is ERC721, Ownable {
          We circumvent the ERC721 transfer function.
     @param vault Vault that needs to get transfered.
   */
-    function _liquidateTransfer(address vault) internal {
+    function _liquidateTransfer(address vault, address liquidator) internal {
+//        (bool success, address liquidator) = IVault(vault).liquidateVault(
```
Comment (Thomas-Smets) [inline src/Factory.sol:363]: Remove comments

File: src/Vault.sol [hunk 4]
```diff
@@ -878,13 +888,13 @@ contract Vault {
          Sets debtInfo todo: needed?
          Transfers ownership of the proxy vault to the liquidator!
     @param liquidationKeeper Addross of the keeper who initiated the liquidation process.
-    @param _liquidator Contract Address of the liquidation logic.
     @return success Boolean returning if the liquidation process is successfully started.
   */
-    function liquidateVault(address liquidationKeeper, address _liquidator)
+
```
Comment (Thomas-Smets) [inline src/Vault.sol:893]: Remove whitespace

File: src/Factory.sol (hunk 1 above)
Comment (alikonuk1) [reply src/Factory.sol]: Removed

File: src/Factory.sol (hunk 2 above)
Comment (alikonuk1) [reply src/Factory.sol]: Removed

File: src/Factory.sol (hunk 3 above)
Comment (alikonuk1) [reply src/Factory.sol]: Removed

File: src/Vault.sol (hunk 4 above)
Comment (alikonuk1) [reply src/Vault.sol]: Removed

File: src/Vault.sol [hunk 5]
```diff
@@ -870,6 +870,16 @@ contract Vault {
                           LIQUIDATION LOGIC
     ///////////////////////////////////////////////////////////////*/
 
+    /** 
+    @notice Function to set a new contract for the liquidation logic
+    @dev Since vaults to be liquidated, together with the open debt, are transferred to the protocol,
+         New logic can be set without needing to increment the vault version.
+    @param _liquidator The new liquidator contract
+  */
+    function setLiquidator(address _liquidator) public onlyOwner {
```
Comment (j-vp) [inline src/Vault.sol:879]: to be removed @Thomas-Smets

**Review by j-vp — CHANGES_REQUESTED** [review]

**Review by j-vp — APPROVED** [review]
lgtm

**Pull request merged**